packages/server/src/script/fetchMGErrors.ts

```diff
@@ -0,0 +1,129 @@
+import { Logging } from '@google-cloud/logging'
+import { processEnvOrThrow } from '../common'
+import { MailgunHookBody, MailgunEventReason } from '../service/webhooks'
+import { promises as fs } from 'fs'
+
+interface EntryResponse {
+  data: MailgunHookBody
+}
+
+/**
+ * * The first is the starting date (inclusive) of the script;
+ * * Likewise, the second can be the end date (inclusive), or null if we
+ * want to fetch all records from the starting point until the present moment;
+ * * The third is a list of MGEventReason to be ignored while fetching logs;
+ */
+export type MGFetchErrorsArgs = [Date, Date | null, MailgunEventReason[]]
+// Only used to ease importing through require()
+interface MGFetchErrorsArgsExport { args: MGFetchErrorsArgs }
+
+const { args } = process.env.CI
+  ? require('./data/mgConditions.sample.ts') as MGFetchErrorsArgsExport
+  : require('./data/mgConditions.nogit.ts') as MGFetchErrorsArgsExport
+
+const projectId = processEnvOrThrow('GCLOUD_PROJECT')
+
+const fetchMGErrors = async (
+  fromIsoStamp: string,
+  untilIsoStamp: string | null,
+  ignoredEventReasons: MailgunEventReason[] = [],
+) => {
+  const logging = new Logging({ projectId })
+
+  const filter =
+`logName="projects/${projectId}/logs/mailgun-log"
+severity=(WARNING OR ERROR OR CRITICAL OR ALERT OR EMERGENCY)
+
+timestamp >= "${fromIsoStamp}"
+${untilIsoStamp ? `timestamp <= ${untilIsoStamp}` : ''}`
+
+  console.log('Querying all MG errors in our logs during the period, this might take a while...\n')
+  // The response comes in a nested array, flattening it to ease iterations
+  const entries = (await logging.log('mailgun-log').getEntries({ filter })).flat() as EntryResponse[]
+
+  const results: Record<string, string[]> = {}
+
+  entries.forEach(({ data }) => {
+    // Do not resend since the email doesn't exist, apparently the message
+    // for this event vary depending on the destination
+    //
+    // We don't use the delivery-status code here since email providers'
+    // implementation differ slightly from each other, for example at Google
+    // Gmail we have 550 for non existent emails, but on Yahoo the same error
+    // is represented by 554.
+    const message = data['event-data']?.['delivery-status']?.message
+
+    // Message can only be undefined/null in staging/dev environments, this
+    // happens when we've used MG's webhook testing hooks before (they don't
+    // contain delivery-statuses)
+    if (!message) {
+      return
+    }
+
+    // Found on production logs
+    if (message.indexOf('No SMTP service') >= 0) {
```

Comment:
What does this mean and why do we ignore it? (add comment in code)

Reply:
Removed this condition, it's from mail.com and apparently this same message can be used when they're blocking us from reaching their inbox (like when they considered our old MG IP as spam)